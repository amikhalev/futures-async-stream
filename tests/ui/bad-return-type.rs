#![feature(async_await, generators)]

use futures_async_stream::async_stream;

#[async_stream(item = Option<i32>)]
async fn foo() -> i32 {} //~ ERROR async stream functions must return the unit type

#[async_stream(item = (i32, i32))]
async fn unit() -> () {} // OK

fn main() {}
