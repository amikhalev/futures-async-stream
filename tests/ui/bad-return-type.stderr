error: async stream functions must return the unit type
 --> $DIR/bad-return-type.rs:6:19
  |
6 | async fn foo() -> i32 {} //~ ERROR async stream functions must return the unit type
  |                   ^^^

error: aborting due to previous error

